tiatoolbox/models/engine/patch_classifier.py

```diff
@@ -0,0 +1,553 @@
+"""Defines PatchClassifier Engine."""
+
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import zarr
+from typing_extensions import Unpack
+
+from .engine_abc import EngineABCRunParams
+from .patch_predictor import PatchPredictor
+
+if TYPE_CHECKING:  # pragma: no cover
+    import os
+    from pathlib import Path
+
+    import numpy as np
+
+    from tiatoolbox.annotation import AnnotationStore
+    from tiatoolbox.models.engine.io_config import ModelIOConfigABC
+    from tiatoolbox.models.models_abc import ModelABC
+    from tiatoolbox.wsicore import WSIReader
+
+
+class ClassifierRunParams(EngineABCRunParams):
+    """Class describing the input parameters for the :func:`EngineABC.run()` method.
+
+    Attributes:
+        batch_size (int):
+            Number of image patches to feed to the model in a forward pass.
+        cache_mode (bool):
+            Whether to run the Engine in cache_mode. For large datasets,
+            we recommend to set this to True to avoid out of memory errors.
+            For smaller datasets, the cache_mode is set to False as
+            the results can be saved in memory.
+        cache_size (int):
+            Specifies how many image patches to process in a batch when
+            cache_mode is set to True. If cache_size is less than the batch_size
+            batch_size is set to cache_size.
+        class_dict (dict):
+            Optional dictionary mapping classification outputs to class names.
+        device (str):
+            Select the device to run the model. Please see
+            https://pytorch.org/docs/stable/tensor_attributes.html#torch.device
+            for more details on input parameters for device.
+        ioconfig (ModelIOConfigABC):
+            Input IO configuration (:class:`ModelIOConfigABC`) to run the Engine.
+        return_labels (bool):
+            Whether to return the labels with the predictions.
+        num_loader_workers (int):
+            Number of workers used in :class:`torch.utils.data.DataLoader`.
+        num_post_proc_workers (int):
+            Number of workers to postprocess the results of the model.
+        output_file (str):
+            Output file name to save "zarr" or "db". If None, path to output is
+            returned by the engine.
+        patch_input_shape (tuple):
+            Shape of patches input to the model as tuple of height and width (HW).
+            Patches are requested at read resolution, not with respect to level 0,
+            and must be positive.
+        resolution (Resolution):
+            Resolution used for reading the image. Please see
+            :class:`WSIReader` for details.
+        return_labels (bool):
```

Comment:
repeated argument

Reply:
maybe remove the one above

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/pull/865/commits/421ea775e7c7baef13caaba9ecac90e987c19f57